project-koku/koku
# Catch all message processing and provider sync exceptions in sources client

There has been a few times where the sources event loop has stopped working due to an uncaught exception.  This change will catch all exceptions while message processing and synchronizing source-providers to ensure that the event loop remains operational at all times.

FYI: This was done on the OCP listener here: https://github.com/project-koku/koku/blob/2ea029b48cbd0d746343f95dfa739d4d0271c5d7/koku/masu/external/kafka_msg_handler.py#L327

**Testing**
1. Add some temporary code to force exceptions at specific points and to verify that the event loop remains healthy
2. Add a cost management source (source_id 1) and verify it's successful
3.  Add a cost management source (source_id 2) and veirfy it's not successful from an exception thrown in `sources_network_info`
4.  Add a cost management source (source_id 3) and verify that it's successful (showing the event loop is still operational)
5.  Add a cost management source (source_id 4) and verify it's not successful from an exception thrown in execute_koku_provider_op while syncing with koku_fork
6. Add a final cost management source (source_id 5) and verify it's successful showing the event loop is still operational

**Test Results**
[sources_hardening_ut.txt](https://github.com/project-koku/koku/files/3815822/sources_hardening_ut.txt)
